charts/sourcegraph/templates/_helpers.tpl

```diff
@@ -41,7 +41,16 @@ Create the name of the service account to use
 {{- define "sourcegraph.serviceAccountName" -}}
 {{- $top := index . 0 }}
 {{- $service := index . 1 }}
-{{- default $service (index $top.Values $service "serviceAccount" "name") }}
+{{- $defaultServiceAccountName := ((snakecase $service) | replace "_" "-") }}
```

Comment:
I think the convention of resource name is dashed instead of camelCase

this line does the following 

camelCase -> snake_case -> replace("_", "-") 

repoUpdater -> repo_updater -> repo-updater

codeInsightsDb -> code_insights_db -> code-insights-db
in this case, ideally we want `codeinsights-db` to be consistent with the deployment name, but the top-level key in `values.yaml` is `codeInsightsDB` instead of `codeinsightsDB.